tests/unit/lms/views/lti/deep_linking_test.py

```diff
@@ -41,3 +61,64 @@ def context(self):
         context = mock.create_autospec(LTILaunchResource, spec_set=True, instance=True)
         context.js_config = mock.create_autospec(JSConfig, spec_set=True, instance=True)
         return context
+
+
+class TestDeepLinkingFieldsView:
+    @pytest.mark.usefixtures("application_instance_service")
+    @pytest.mark.parametrize(
+        "content,extra_params,url",
+        [
+            (
+                {"type": "file", "file": {"id": 1}},
+                {},
+                "http://example.com/lti_launches?canvas_file=true&file_id=1",
+            ),
+            (
+                {"type": "url", "url": "https://example.com/content"},
+                {},
+                "http://example.com/lti_launches?url=https%3A%2F%2Fexample.com%2Fcontent",
+            ),
+            (
+                {"type": "url", "url": "https://example.com/content"},
+                {"extra": "value"},
+                "http://example.com/lti_launches?extra=value&url=https%3A%2F%2Fexample.com%2Fcontent",
+            ),
+        ],
+    )
+    def test_it(
+        self,
+        content,
+        extra_params,
+        url,
+        ltia_http_service,
+        application_instance,
+        pyramid_request,
+    ):
+        pyramid_request.parsed_params.update(
+            {
+                "content": content,
+                "extra_params": extra_params,
+            }
+        )
+
+        fields = DeepLinkingFieldsViews(pyramid_request).file_picker_to_formfields_v13()
+
+        ltia_http_service.sign.assert_called_once_with(
+            {
+                "https://purl.imsglobal.org/spec/lti/claim/deployment_id": application_instance.deployment_id,
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingResponse",
+                "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+                "https://purl.imsglobal.org/spec/lti-dl/claim/content_items": [
+                    {"type": "ltiResourceLink", "url": url}
+                ],
+                "https://purl.imsglobal.org/spec/lti-dl/claim/data": sentinel.deep_linking_settings,
+            }
+        )
+        assert fields["JWT"] == ltia_http_service.sign.return_value
+
+    @pytest.fixture
+    def pyramid_request(self, pyramid_request):
+        pyramid_request.parsed_params = {
+            "deep_linking_settings": sentinel.deep_linking_settings
+        }
+        return pyramid_request
```

Comment:
It just occurred to me it would be really sweet if matchers captured the last thing they matched:

```python
expected_url = Any.url()
assert {"some_big_thing" : expected_url}
assert expected_url.last_match == ...
```

That way you could make kind of vague statements about the whole, then make more specific ones about parts you snip out after.